src/main/java/report/LoadStock.java

```diff
@@ -0,0 +1,211 @@
+package report;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import exceptions.InvalidFilePathException;
+import exceptions.StockReadWriteException;
+import ingredient.Ingredient;
+import stock.Stock;
+
+/**
+ * This class encapsulates the functionality of loading data related to
+ * the 'Stock' category in the report.txt file.
+ *
+ */
+public class LoadStock {
+    
+    /** The default file name that is to be read or written by the program. */
+    public static final String DEFAULT_STORAGE_FILEPATH = "report.txt";
+    
+    /** Constructor of a Storage using the default file path. */
+    public final Path path;
+    
+    /** 
+     * The relative index to start decoding for price. In 'PRICE:$0.11';
+     * index 7 is that start of the price to be parsed. 
+     */
+    private final int startIndexToDecodeForPrice = 7;
+    
+    /** 
+     * The relative index to start decoding for price. In 'Quantity:10';
+     * index 9 is that start of the price to be parsed. 
+     */   
+    private final int startIndexToDecodeForQuantity = 9;
+
+    /** 
+     * Constructor of a LoadStock retrieved from Path.
+     * @throws InvalidStorageFilePathException If the report data file does not match
+     *                                         a .txt extension.
+     */
+    public LoadStock() {
+        this(DEFAULT_STORAGE_FILEPATH);
+    }
+
+    public LoadStock(String filePath) {
+        this.path = Paths.get(filePath);
```

Comment:
javadoc seems incorrect here. The javadoc suppose belong to the constructor below? And it seems that the exception typo?